### ISSUE TITLE
[bug] assertion failed at ..\level_aux_mesh.cxx:94

### ISSUE THREAD

**Issue by Wolf-Heart** [abramcumner/xray_re-tools#13]
Не декомпелируется локация la01_escape из Lost Alpha. Конвертер выдает ошибку:

```
saving error list
building HOM object
building SOM object
[bug] assertion failed at ..\level_aux_mesh.cxx:94
```

А точнее, не создаются: папка la01_escape и файл la01_escape.level в папке: editors\rawdata\levels
https://i.postimg.cc/j2WLtVVd/1.png
В то время как в папке: editors\rawdata\objects\levels\la01_escape объекты декомпелированы:
https://i.postimg.cc/Cxj17HH5/2.png
Попробовал декомпелировать другой уровень la15_darkscape, декомпил прошел успешно и все необходимые файлы были созданы:
https://i.postimg.cc/sgvxWQwH/3.png
Не знаю, может еще есть такие уровни, которые не декомпилятся, буду тестить, а пока прилагаю полные логи декомпила la01_escape и la15_darkscape: https://yadi.sk/d/jwuzZzU37h3Imw
А так-же на всякий случай уровень la01_escape: https://yadi.sk/d/fhjRrT4RjnewXg
Если еще что необходимо, пиши предоставлю.

**Comment by abramcumner** [comment]
Не найден материал для SOM.

**Comment by Wolf-Heart** [comment]
la15_darkscape хоть и декомпелировался вроде успешно, но при попытке загрузить уровень в СДК происходит мгновенный без логовый вылет.
Скорее всего дело в том, что файлы в папке editors\rawdata\levels\la15_darkscape создались все бинарными, в то время как в СДК_ЗП они пактически все текстовые.
А тут они даже не совсем бинарные, гибрид какой-то, полу бинарные, полу текстовые:
https://i.postimg.cc/W4BRWmhd/1.png
Прилагаю папку levels: https://yadi.sk/d/ETretlOaq4ExYA

**Comment by Wolf-Heart** [comment]
> Не найден материал для SOM. 
Это проблема конвертера? Или...???

**Comment by abramcumner** [comment]
> Скорее всего дело в том, что файлы в папке editors\rawdata\levels\la15_darkscape создались все бинарными, в то время как в СДК_ЗП они пактически все текстовые.

Наверное ты декомпилировал в СДК ТЧ, поэтому в СДК ЗП и не открываются.

**Comment by abramcumner** [comment]
> 
> 
> > Не найден материал для SOM.
> 
> Это проблема конвертера? Или...???

Это проблема gamemtl.xr в СДК, в нет подходящего материала.

**Comment by Wolf-Heart** [comment]
> Наверное ты декомпилировал в СДК ТЧ, поэтому в СДК ЗП и не открываются.

Улыбнуло!
Я что по твоему совсем с дуба рухнул?   :)))

**Comment by Wolf-Heart** [comment]
> Это проблема gamemtl.xr в СДК, в нет подходящего материала

Хм... А по чему с предыдущей версией например 0.1.11 такого не было?
Я ей успешно декомпилил эти уровни и проблем таких не было.

**Comment by abramcumner** [comment]
В прошлой версии наверное настроил секцию la_config и префикс la указывал. С этой чего-то не сделал. Скинь converter.ini и строку запуска.

**Comment by Wolf-Heart** [comment]
> В прошлой версии наверное настроил секцию la_config

В этой, то-же все настроил. Но интересное кино получается...
Я декомпилил с ключем la: в батнике, и получился такой результат. А после твоего комита:


> Наверное ты декомпилировал в СДК ТЧ, поэтому в СДК ЗП и не открываются.

Решил попробовать с ключем cop: и декомпил la15_darkscape прошел успешно и файлы создались как положенно в текстовом формате:
https://i.postimg.cc/vmB1KJZT/1.png
Сейчас попробую загрузить уровень и потом попробую декомпельнуть la01_escape. По результату отпишусь.

**Comment by Wolf-Heart** [comment]
Так, darkscape загрузился, но при подгрузке текстур, вылезло 5 ошибок:
https://i.postimg.cc/N0Pz2CGS/1.png
https://i.postimg.cc/jSxggYLB/2.png
https://i.postimg.cc/FKcL07hv/3.png
https://i.postimg.cc/RVKhP2j1/4.png
https://i.postimg.cc/sXYHHj4d/5.png
После каждой нажимал ОК, после последней произошел вылет СДК без лога.

**Comment by Wolf-Heart** [comment]
Декомпил la01_escape с ключем cop: в батнике, результат тот же что и с ключем la:
Ошибка та же самая, только объктов менше, 9360 с cop: против 11547 с la:
converter.ini и батник запуска: https://yadi.sk/d/RTCKJtUFldMgRA
Причем изначально секция la в converter.ini была настроенна на папку с распакованной gamedata от LA, а в батнике использовался ключ la:

**Comment by Wolf-Heart** [comment]
>  0.1.11 такого не было

Ошибочка!!! 
Попробовал converter_A.0.1.11 декомпельнуть la01_escape, ошибка та же что и с converter_A.0.2.1
Вот лог, converter.ini и батник запуска: https://yadi.sk/d/FIWCUkGz7ZBQaA
Причем converter.ini и батник настроенны так как были изначально настроенны в converter_A.0.2.1

**Comment by abramcumner** [comment]
> P.S. На название СДК не обращай внимания, это СоР СДК, не SoC - это точно! :)

Какое СДК у тебя не важно, важно что прописано в параметре spawn_version в converter.ini. Вот из твоего:
```
[la_config]:2947_config
$game_data$             = E:\W_w_f_SDK\editors\gamedata\
$game_levels$           = E:\W_w_f_SDK\editors\gamedata\levels\
inadhesive_shaders      = la_inadhesive_shaders
```
Наследуется от ТЧ,  spawn_version не указан => будет конвертировать в формат ТЧ СДК.

> Так, darkscape загрузился, но при подгрузке текстур, вылезло 5 ошибок:
https://i.postimg.cc/N0Pz2CGS/1.png

Ошибка одна, не найден какой-то партикл

**Comment by Wolf-Heart** [comment]
> Наследуется от ТЧ, spawn_version не указан => будет конвертировать в формат ТЧ СДК.

Да, но изначально было прописано:
```
[la_config]:2947_config
$game_data$             = D:\Game_S.T.A.L.K.E.R_Mod\S.T.A.L.K.E.R. - Lost Alpha\UNgamedata\
$game_levels$           = D:\Game_S.T.A.L.K.E.R_Mod\S.T.A.L.K.E.R. - Lost Alpha\UNgamedata\levels\
inadhesive_shaders      = la_inadhesive_shaders
spawn_version = cop
```
Погляди в converter_A.0.1.11: https://yadi.sk/d/FIWCUkGz7ZBQaA
Там те-же самые настройки, что были в converter_A.0.2.1, это я уже потом убрал spawn_version, а результат одинаковый.

**Comment by abramcumner** [comment]
В этом логе тоже не найден материал для SOM.

**Comment by Wolf-Heart** [comment]
Да, я же писал, что результат одинаковый. Именно с этой ошибки и с такими настройками я и начинал декомпилить только сонвертером A.0.2.1. А уже потом начал эксперементировать. :)

**Comment by Wolf-Heart** [comment]
Попробовал декомпилить Кордон из S.T.A.L.K.E.R. - Lost Alpha DC обоими конвертерами.
converter_A.0.2.1 выдал ошибку:
`[bug] assertion failed at C:\projects\xray-re-tools\sources\xray_re\xr_entity.cxx:230`
https://i.postimg.cc/T31kWvSJ/1.png
А converter_A.0.1.11:
`[bug] assertion failed at ..\xr_entity.cxx:230`
https://i.postimg.cc/7Ykh6x5j/2.png
Вот все данные по обоим конвертерам: https://yadi.sk/d/7RrWR3bCji5ofg

**Comment by abramcumner** [comment]
Похоже у них сильно изменен формат нет-пакетов или в спавне начали сохранять клиентские данные.

**Comment by Wolf-Heart** [comment]
Вобщем, интересное кино получается. Декомпелируется la01_escape только из gamedata-ы, которая в SDK_LA. Не помню уже от куда она у меня, но не из LA 1.3003, не из LADC Кордон не декомпелируется. Проверял обоими конвертерами. 
А из gamedata-ы в SDK_LA, декомпилится, но при подгрузке текстур вылазят все те же ошибки о которых писал в комменте выше, а потом вылет без лога.
И причина не партиклы, я удалил файл ps.part, ошибка с партиклом ушла, но вылет все равно произошел после ряда ошибок типа: 
`Please notify AlexMX!!! Critical error has occured in render routine!!! [Type B] - Tools: 'scene_object' Object: 'levels\la01_escape\la01_escape_part2822'`
Это и есть основная причина вылета.
